enterprise/utils.py

```diff
@@ -1108,3 +1110,103 @@ def get_platform_logo_url():
         settings.LMS_ROOT_URL,
         get_logo_url()
     ) if get_logo_url else 'http://fake.url'
+
+
+def create_tableau_user(username, enterprise_customer_user):
+    """
+    Create the user in tableau and store the tableau user if in the EnterpriseAnalyticsUser model.
+    """
+    from enterprise.models import EnterpriseAnalyticsUser  # pylint: disable=import-outside-toplevel
+    tableau_auth, server = get_tableau_server()
+    if tableau_auth and server and enterprise_customer_user:
+        enterprise_uuid = enterprise_customer_user.enterprise_customer.uuid
+        try:
+            with server.auth.sign_in(tableau_auth):
+                user_item = TSC.UserItem(username, TSC.UserItem.Roles.Viewer)
+                user = server.users.add(user_item)
+                enterprise_analytics_user = EnterpriseAnalyticsUser.objects.create(
+                    enterprise_customer_user=enterprise_customer_user,
+                    analytics_user_id=user.id
+                )
+                enterprise_analytics_user.save()
+                add_user_to_tableau_group(user.id, str(enterprise_uuid))
+        except ServerResponseError as exc:
+            LOGGER.error(
+                '[TABLEAU USER SYNC] Could not sync enterprise admin user %s in tableau.'
+                'Server returned: %s', username, str(exc),
+            )
+
+
+def get_tableau_server():
+    """
+    Return Tableau ServerAuth and Server instances or None
+    """
+    try:
+        tableau_auth = TSC.TableauAuth(settings.TABLEAU_ADMIN_USER, settings.TABLEAU_ADMIN_USER_PASSWORD)
+        server = TSC.Server(settings.TABLEAU_URL)
+        return tableau_auth, server
+    except AttributeError as err:
+        LOGGER.error(
+            '[TABLEAU USER SYNC] Could not sync enterprise admin user in tableau.'
+            'Error detail: %s', str(err),
+        )
+        return None, None
+
+
+def add_user_to_tableau_group(tableau_user_id, tableau_group):
+    """
+    Associate a user with the group in tableau. Create the group if it does not already exist.
+    """
+    tableau_auth, server = get_tableau_server()
+    if tableau_auth and server:
+        try:
+            with server.auth.sign_in(tableau_auth):
+                group_to_create = TSC.GroupItem(tableau_group)
+                group = server.groups.create(group_to_create)
+                server.groups.add_user(group, tableau_user_id)
```

Comment:
I am not sure what this flow is doing? Can we add a link to docs here for reviewers and future contributors to help understand what this flow is doing and why we need it?

Reply:
Added explanation.